Added ResponseEntity at Controller

simon-1729 opened the pull request:
Added spring ResponseEntity for one method in Controller.

CodeReaper9 committed:
Added ResponseEntity at Controller

CodeReaper9 commented on src/main/java/com/modernjava/demo/controller/ModernJavaController.java:
```diff
@@ -37,7 +41,10 @@ public String filtered(@PathVariable String filterType) {
 	@GetMapping(path = "/filteredJ8/{filterType}")
-	public String filteredJ8(@PathVariable String filterType) {
-		return appleService.filterApplesJ8(filterType);
+	public ResponseEntity<List<Apple>> filteredJ8(@PathVariable
```
What about the other methods in the controller that are still returning String. It might be idea to update to return ResponseEntity too.
Also, if you give an invalid path parameter you get back the unfiltered and unsorted list. Is this correct behavior would it be better to return an error message? And, provide an additional end-point should the client wish to see unadulterated list.

simon-1729 replied:
Thanks for the suggestion, yes I agree.
I'll update the pull request.

CodeReaper9 committed:
Added unit test

CodeReaper9 committed:
Updated Controller, Added Tests.

All end points return ResponseEntity.
Added Test for methods in service layer.
Filter by weight filters all apples greater
than HEAVY_APPLE limit.

CodeReaper9 reviewed (approved):
Looks good! :+1: 
Only, suggestion would be to add a logger.
